### PR TITLE
Support building this repo for more platforms, by checking the build triple

### PR DESCRIPTION
**Pull request by buttaface**
~This is needed for all platforms that don't have multi-architecture libraries like Darwin. Also, add the new architecture-specific rpath to the resulting swift-driver and make it possible to cross-compile this repo for non-Darwin platforms on Darwin.~

~The driver changes are the Swift translation of the Driver-specific changes of apple/swift#63782.~

~Resolves apple swift#63645~

This replaces all but one prior check of the host system instead, plus removed
unnecessary regex checking.

### CODE REVIEW

File: Utilities/build-script-helper.py [hunk 1]
```diff
@@ -595,10 +599,14 @@ def cmake_build(args, swiftc_exec, cmake_args, swift_flags, source_path,
...
-def get_build_target(swiftc_path, args):
+def get_build_target(swiftc_path, args, cross_compile=False):
     """Returns the target-triple of the current machine."""
     try:
-        target_info_json = subprocess.check_output([swiftc_path, '-print-target-info'],
+        command = [swiftc_path, '-print-target-info']
+        if cross_compile:
+            cross_compile_json = json.load(open(args.cross_compile_config))
+            command += ['-target', cross_compile_json["target"]]
+        target_info_json = subprocess.check_output(command,
```
Comment (buttaface) [inline Utilities/build-script-helper.py:609]: I copied this over from [my earlier change to the SPM bootstrap script](https://github.com/apple/swift-package-manager/commit/38df7c37e7ff705c6f48544a28d3d7a4839e290f#diff-fdc89d912e967e4b700e1bb41e08552bb416b5c5551bbae5b1ccac134db6c3e5R287).

**Comment by buttaface** [comment]
Updated this pull with build script changes to allow cross-compiling to Android on macOS: @artemcm, let me know what you think of the build script changes alone for now.

**Comment by buttaface** [comment]
@artemcm, now that this passed the CI on the linked compiler pull, looking for review so I can get this in before the upcoming 5.9 branch. Please don't merge though, as all 11 pulls will have to be merged together.

**Comment by buttaface** [comment]
Removed the architecture-specific changes since [the Swift devs want to take a different approach](https://github.com/apple/swift/pull/63782#pullrequestreview-1353429744), leaving only these build script modifications I had also made.

One last CI run and this can go in, @artemcm.

**Comment by buttaface** [comment]
Ping @artemcm, this pull look okay to you?

**Review by artemcm — APPROVED** [review]

**Comment by buttaface** [comment]
Passed all CI, ready for merge.

**Pull request merged**